FRAuth/FRAuth/Callbacks/DeviceSigningVerifierCallback.swift

```diff
@@ -0,0 +1,243 @@
+                throw AuthError.invalidCallbackResponse("Failed to parse output")
+            }
+            outputDictionary[outputName] = outputValue
+        }
+        
+        self.userId = outputDictionary[CBConstants.userId] as? String
+        
+        guard let challenge = outputDictionary[CBConstants.challenge] as? String else {
+            throw AuthError.invalidCallbackResponse("Missing challenge")
+        }
+        self.challenge = challenge
+        
+        guard let title = outputDictionary[CBConstants.title] as? String else {
+            throw AuthError.invalidCallbackResponse("Missing title")
+        }
+        self.title = title
+        
+        guard let subtitle = outputDictionary[CBConstants.subtitle] as? String else {
+            throw AuthError.invalidCallbackResponse("Missing subtitle")
+        }
+        self.subtitle = subtitle
+        
+        guard let promptDescription = outputDictionary[CBConstants.description] as? String else {
+            throw AuthError.invalidCallbackResponse("Missing description")
+        }
+        self.promptDescription = promptDescription
+        
+        self.timeout = outputDictionary[CBConstants.timeout] as? Int
+        
+        //parse inputs
+        var inputNames = [String]()
+        for input in inputs {
+            guard let inputName = input[CBConstants.name] as? String else {
+                throw AuthError.invalidCallbackResponse("Failed to parse input")
+            }
+            inputNames.append(inputName)
+        }
+        
+        guard let jwsKey = inputNames.filter({ $0.contains(CBConstants.jws) }).first else {
+            throw AuthError.invalidCallbackResponse("Missing jwsKey")
+        }
+        self.jwsKey = jwsKey
+        
+        guard let clientErrorKey = inputNames.filter({ $0.contains(CBConstants.clientError) }).first else {
+            throw AuthError.invalidCallbackResponse("Missing clientErrorKey")
+        }
+        self.clientErrorKey = clientErrorKey
+        
+        try super.init(json: json)
+        type = callbackType
+        response = json
+    }
+    
+    
+    /// Sign the device.
+    /// - Parameter completion Completion block for Device binding result callback
+    open func sign(completion: @escaping DeviceBindingResultCallback) {
+        execute(userKeyService: nil, completion)
+    }
+    
+    
+    /// Helper method to execute binding , signing, show biometric prompt.
+    /// - Parameter userKeyService: service to sort and fetch the keys stored in the device
+    /// - Parameter completion: completion Completion block for Device binding result callback
+    internal func execute(userKeyService: UserKeyService?,
+                          _ completion: @escaping DeviceBindingResultCallback) {
+        let newUserKeyService = userKeyService ?? UserDeviceKeyService(encryptedPreference: nil)
```

Comment:
DeviceBindingResultCallback ? can we alias to DeviceSigningResultCallback

Reply:
Created DeviceSigningResultCallback and used that in DeviceSigningVerifierCallback